frontend/packages/knative-plugin/src/components/add/EventSourceForm.tsx

```diff
@@ -30,29 +33,55 @@ const EventSourceForm: React.FC<FormikProps<FormikValues> & OwnProps> = ({
   dirty,
   namespace,
   projects,
-}) => (
-  <Form className="co-deploy-image" onSubmit={handleSubmit}>
-    <EventSourcesSelector eventSourceList={useEventSourceList(namespace)} />
-    {values.type === EventSources.CronJobSource && <CronJobSection />}
-    {values.type === EventSources.SinkBinding && <SinkBindingSection />}
-    {values.type === EventSources.ApiServerSource && <ApiServerSection />}
-    {values.type === EventSources.KafkaSource && <KafkaSourceSection />}
-    <SinkSection namespace={namespace} />
-    <AppSection
-      project={values.project}
-      noProjectsAvailable={projects?.loaded && _.isEmpty(projects.data)}
-    />
-    {values.type === EventSources.KafkaSource && <AdvancedSection />}
-    <FormFooter
-      handleReset={handleReset}
-      errorMessage={status && status.submitError}
-      isSubmitting={isSubmitting}
-      submitLabel="Create"
-      sticky
-      disableSubmit={!dirty || !_.isEmpty(errors)}
-      resetLabel="Cancel"
-    />
-  </Form>
-);
+}) => {
+  const EventSourceSection: React.FC = renderEventSourceSection(values.type);
+  const { setFieldValue, setFieldTouched, validateForm } = useFormikContext<FormikValues>();
+  const [namespaceField] = useField('project.name');
+  const handleEventSourceChange = React.useCallback(
```

Comment:
Why have you moved this handler out of the selector component and into this parent form component?

Reply:
moved handler function to `EventSourceSelector`